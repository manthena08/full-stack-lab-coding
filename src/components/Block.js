import React from 'react'
import { Box, Typography, makeStyles } from '@material-ui/core';
import PropTypes from "prop-types";

const useStyles = makeStyles((theme) => ({
    root: {
       backgroundColor: '#e0e0e0',
       padding: theme.spacing(1),
       marginBottom: theme.spacing(1)
    },
    orderColor: {
        color: '#8a9aee'
    }

}));
const Block = (props) => {
    const classes = useStyles();
    const getBlockOrders = (order) => {
        if(order > 99){
            return order;
        }else if (order > 9){
            return String(order).padStart(3, '0')
        }else{
            return String(order).padStart(3, '00')
        }
    }
    return (
        <Box className={classes.root}>
            <Typography className={classes.orderColor}>{getBlockOrders(props.block.attributes.index)}</Typography>
            <Typography>{props.block.attributes.data}</Typography>

        </Box>
    )
}

Block.propTypes = {
    block: PropTypes.object.isRequired
};

export default Block;