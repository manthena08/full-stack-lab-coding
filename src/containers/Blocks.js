import React, { useEffect } from 'react'
import PropTypes from "prop-types";
import { Box } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import { getBlocksByNodeUrl, getBlockDetailsSelector } from '../actions/blocks';
import Block from '../components/Block';

const Blocks = (props) => {
    const dispatch = useDispatch();
    const blocksData = useSelector(state => getBlockDetailsSelector(state.blocks, props.nodeUrl));

    useEffect(() => {
        dispatch(getBlocksByNodeUrl(props.nodeUrl))
    }, [])

    return (
        <Box width="100%">
            {
                blocksData.loading &&
                <Box>
                    loading...
                </Box>
            }
            {
                !blocksData.loading && !blocksData.success &&
                <Box>
                    Failed to get the data. Please try again.
                </Box>
            }
            {
                blocksData.data.map(item => (
                    <Block key={item.key} block={item} />
                ))
            }
        </Box>
    )
}


Blocks.propTypes = {
    nodeUrl: PropTypes.string.isRequired
};

export default Blocks;