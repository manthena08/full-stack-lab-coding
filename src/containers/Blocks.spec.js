import React from 'react';
import thunk from 'redux-thunk';
import configureMockStore from "redux-mock-store";
import { create } from "react-test-renderer";
import Blocks from './Blocks';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import Block from '../components/Block';

describe('Blocks', () => {
    const blocks = {
        list: [
            {
                url: 'sample',
                loading: false,
                succes: false,
                data: [{
                    attributes:{
                        index: 1
                    }
                }]
            }
        ]
    }

    it("check block element exist", () => {
        const middlewares = [thunk];
        const store = configureMockStore(middlewares)({blocks});
        const component = mount(
            <Provider store={store}>
                <Blocks nodeUrl="sample"  />
            </Provider>
        );
        expect(component.find(Block).length).toEqual(1)
        
    });

    it("should match snapshot", () => {
        const middlewares = [thunk];
        const store = configureMockStore(middlewares)({blocks});
        const component = create(
            <Provider store={store}>
                <Blocks nodeUrl="sample"/>
            </Provider>
        );
        const tree = component.toJSON();

        expect(tree).toMatchSnapshot();
    });
})