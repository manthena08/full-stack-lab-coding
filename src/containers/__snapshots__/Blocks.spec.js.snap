// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Blocks should match snapshot 1`] = `
<div
  className="MuiBox-root MuiBox-root-4"
>
  <div
    className="MuiBox-root MuiBox-root-5"
  >
    Failed to get the data. Please try again.
  </div>
</div>
`;
